app/models/manageiq/providers/hawkular/inventory/operation_notification.rb

```diff
@@ -12,7 +12,7 @@ def initialize(args, manager)
 
           def emit
             ActiveRecord::Base.connection_pool.with_connection do
-              mw_entity = args.entity_klass.find_by(:id => args.target_resource) unless args.entity_klass == MiddlewareServer
+              mw_entity = args.entity_klass.find_by(:ems_ref => args.target_resource) unless args.entity_klass == MiddlewareServer
```

Comment:
This line is actually fixing notification parsing for all cases when the `ems_ref` was not `MiddlewareServer` (all operations which does not go via MiddlewareServer controler), such as:
- remove MW datasource
- undeploy, disable, enable, restart of MW deployment

Reply:
@abonas, do we have a ticket for that ^ as well? Because without this change no notifications are posted for MW deployment operations neither...

Reply:
@tumido not afaik. if it's needed, please open a bz so qe can understand the full context of the fix that applies to other entities too.

Reply:
@abonas, I hope I did it right: [BUG 1518292](https://bugzilla.redhat.com/show_bug.cgi?id=1518292)

Reply:
Perhaps, could you add a small test to check that the notification is placed? (At least, for the _remove datasource_ operation that you are fixing)

I guess it may look like these ones:
https://github.com/ManageIQ/manageiq-providers-hawkular/blob/f02dd94d3b248878d18d5a2287fa0a1b56373414/spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb#L176-L186